packages/store-ui/src/molecules/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,197 @@
+import type { PropsWithChildren } from 'react'
+import React, { useMemo } from 'react'
+import type { SwipeableProps } from 'react-swipeable'
+
+import Button from '../../atoms/Button'
+import Icon from '../../atoms/Icon'
+import { RightArrowIcon, LeftArrowIcon } from './Arrows'
+import useSlider from '../../hooks/useSlider/useSlider'
+import useSlideVisibility from './hooks/useSlideVisibility'
+import Bullets from '../Bullets'
+
+export interface CarouselProps extends SwipeableProps {
+  testId?: string
+  infiniteMode?: boolean
+  controls?: 'complete' | 'navigationArrows' | 'paginationBullets'
+  transition?: {
+    duration: number
+    property: string
+    delay?: number
+    timing?: string
+  }
+}
+
+const createTransformValues = (infinite: boolean, totalItems: number) => {
+  const transformMap: Record<number, number> = {}
+  const slideWidth = 100 / totalItems
+
+  for (let idx = 0; idx < totalItems; ++idx) {
+    const currIdx = infinite ? idx - 1 : idx
+    const transformValue = -(slideWidth * idx)
+
+    transformMap[currIdx] = transformValue
+  }
+
+  return transformMap
+}
+
+function Carousel({
+  infiniteMode = true,
```

Comment:
Should the carousel as a molecule be opinionated?

Reply:
Uhm, I wouldn't consider these default values to be opinionated... but rather necessary for the component to work. I mean, we could have no default values for these props, and then users would always have to pass them. But I don't see how that makes this component more or less opinionated. What do you think?

Reply:
Also, I think there's a slite change in how this component has turned out to be. 

Think of this Carousel as not being a generic `slider-layout` equivalent, but more like an actual image carousel. Later on, we're going to have a Shelf that will not use this Carousel. It's only going to use the `useSlider` hook.

I would go as far as saying this is now an Organism 🤔 . And considering the "molecule" part as being the `useSlider` hook. What do you think?

Reply:
Yeah, you're right. I asked because isn't clear enough for me when a component should have this kind of customization, you know!?